dashboard/pages/3_Sentiment_Over_Time.py

```diff
@@ -25,50 +28,59 @@ def resample_dataframe(df: pd.DataFrame, time_interval: str, aggregate: str):
     return pd.DataFrame(df_avg)
 
 
-def construct_streamlit_time_graph(data_df: pd.DataFrame, avg_col, count_col, sent_by_title: bool, sampling: str):
+def construct_streamlit_time_graph(data: pd.DataFrame, avg_col, count_col, sent_by_title: bool, sampling: str):
     """Constructs a streamlit time graph."""
+    averaged = resample_dataframe(data, sampling, "mean").dropna()
 
-    averaged = resample_dataframe(data_df, sampling, "mean").dropna()
-
-    counts = resample_dataframe(data_df, sampling, "count").dropna()
+    counts = resample_dataframe(data, sampling, "count").dropna()
 
     avg_graph = visualise_change_over_time(
         averaged, by_title=sent_by_title)
 
     count_graph = visualise_change_over_time(
         counts, by_title=sent_by_title)
 
-    avg_col.subheader("Average")
+    avg_col.subheader(f"Average")
     avg_col.altair_chart(avg_graph, use_container_width=True)
 
-    count_col.subheader("Count")
+    count_col.subheader(f"Count")
     count_col.altair_chart(count_graph, use_container_width=True)
 
 
-@st.cache_data
-def add_year_month_day_columns(data_df: pd.DataFrame) -> pd.DataFrame:
-    """Adds year, month, and weekday columns to a dataframe"""
-    data_df["year"] = data_df["date_published"].dt.year
-    data_df["month"] = data_df["date_published"].dt.month
-    data_df["weekday"] = data_df["date_published"].dt.day_name()
-    return data_df
+def add_year_month_day_columns(data: pd.DataFrame) -> pd.DataFrame:
```

Comment:
Great idea to split the heatmap up like this! Will make the results really interesting to look at over time.